### PR TITLE
Upgrade to SQLAlchemy 2.0

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR pairs with https://github.com/cesium-ml/baselayer/pull/349 to upgrade to SQLAlchemy 2.0

### CODE REVIEW

File: skyportal/facility_apis/lco.py [hunk 1]
```diff
@@ -539,6 +542,8 @@ def download_observations(request_id, ar):
         session.commit()
     except Exception as e:
         return log(f"Unable to post data for {request_id}: {e}")
+    finally:
+        session.close()
```
Comment (guynir42) [inline skyportal/facility_apis/lco.py:546]: Do we know the difference between `session.close()` and `session.remove()`?

File: skyportal/handlers/api/analysis.py [hunk 2]
```diff
@@ -43,8 +41,6 @@
 
 _, cfg = load_env()
 
-Session = scoped_session(sessionmaker(bind=DBSession.session_factory.kw["bind"]))
```
Comment (guynir42) [inline skyportal/handlers/api/analysis.py:46]: Just don't need this anywhere...?

File: skyportal/handlers/api/observation_plan.py [hunk 3]
```diff
@@ -2078,7 +2078,7 @@ def observation_simsurvey(
         Optional parameters to specify the injection type, along with a list of possible values (to be used in a dropdown UI)
     """
 
-    session = Session()
+    session = Session(bind=DBSession.session_factory.kw["bind"])
```
Comment (guynir42) [inline skyportal/handlers/api/observation_plan.py:2081]: It seems very surprising to me that there would be a difference between setting the `bind` at the `sessionmaker()` level vs. the `Session()` level. Is there any reason given for why this has to be this way in 2.0?

**Review by guynir42 — APPROVED** [review]
If the tests pass, and if we really understand where these differences come from, it looks fine. But I think we should be careful about making changes we don't fully understand.

File: skyportal/facility_apis/lco.py (hunk 1 above)
Comment (mcoughlin) [reply skyportal/facility_apis/lco.py:546]: You close a session = Session() and remove a Session().

File: skyportal/handlers/api/observation_plan.py (hunk 3 above)
Comment (mcoughlin) [reply skyportal/handlers/api/observation_plan.py:2081]: best I can tell they are making it more explicit, and our out of session calls (async calls) requires this).

**Pull request merged**